python/pyarrow/dataset.py

```diff
@@ -714,9 +729,12 @@ def write_dataset(data, base_dir, basename_template=None, format=None,
         and `format` is not specified, it defaults to the same format as the
         specified FileSystemDataset. When writing a Table or RecordBatch, this
         keyword is required.
-    partitioning : Partitioning, optional
+    partitioning : Partitioning or list[str], optional
         The partitioning scheme specified with the ``partitioning()``
-        function.
+        function or a list of field names.
```

Comment:
Maybe mention here that *if* passing a list of field names, you can also specify `partitioning_flavor` in addition.

Reply:
Ping for this one

Reply:
added mention of partitioning_flavor